backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/sourcehu/SourceHUsService.java

```diff
@@ -210,6 +214,25 @@ public boolean isSourceHu(final HuId huId)
 		return Services.get(ISourceHuDAO.class).isSourceHu(huId);
 	}
 
+	public void addSourceHUMarkerIfRequired(@NonNull final HuId huId, @NonNull final ProductId productId, @NonNull final WarehouseId warehouseId)
```

Comment:
I think this should be clearer.
If nothing else, pls add some javadoc here; or improve the method name and e.g. extract `product.isBOM()` into a variable `notAComponent`
Think of a dev wondering why tf it doesn't want to mark a bom-product.

Reply:
PS: also check out the failing test at https://jenkins.metasfresh.com/job/metasfresh/job/gh6987/1/testReport/